src/tests/JIT/Directed/Convert/out_of_range_fp_to_int_conversions.cpp

```diff
@@ -155,16 +155,20 @@ extern "C" DLLEXPORT  uint64_t ConvertDoubleToUInt64(double x, FPtoIntegerConver
         }
 
     case CONVERT_MANAGED_BACKWARD_COMPATIBLE_X86_X64:
-        if (x < int64_max_plus_1)
         {
-            return (x < INT64_MIN) ? (uint64_t)INT64_MIN : (uint64_t)(int64_t)x;
-        }
-        else
-        {
-            x -= int64_max_plus_1;
-            x = trunc(x);
-            return (uint64_t)(((x != x) || (x >= int64_max_plus_1)) ? INT64_MIN : (int64_t)x) + (0x8000000000000000);
+            if (signbit(x) && (x != 0))
+            {
+                return UINT64_MAX;
+            }
+            uint64_t unsigned_bits = *(uint64_t *)&x;
+            int exponent = int((unsigned_bits >> 52) & 0x7ff) - 1023;
+            if (exponent > 63)
+            {
+                return UINT64_MAX;
+            }
+            return (uint64_t)x;
         }
+
```

Comment:
This can probably be removed in favor of `CONVERT_SENTINEL` if the above suggestion to just use the compiler implementation is taken.

Reply:
If it can't, for whatever reason, then comments explaining what the implementation here is doing and why its correct would be beneficial.

MSVC and Clang have different codegen from eachother, but are ultimately doing the "same thing" where they are basically doing a conversion to `int64_t` and then adjusting the result by `2^63` (`9223372036854775808.0`) based on what the input was.

Reply:
This makes sense. ```CONVERT_SENTINEL``` uses the same code as the new one. Changed the test case to use the same for ```x86_x64```